tests/testsuite.src/syn_misc.at

```diff
@@ -4549,6 +4549,250 @@ prog.cob:12: warning: ignoring redundant .
+           PERFORM SEC-2.
+      * Directive purposefully in Area A
+        REPLACE ==RUNNN== BY ==RUN==.
+           STOP RUNNN
+           COPY "sec-1.inc".
+      * Also purposefully in Area A
+        COPY "sec-2.inc".
+])
+
+AT_CHECK([$COMPILE_ONLY prog.cob], [1], [],
+[sec-1.inc: in section 'MAIN':
+sec-1.inc:2: error: 'SEC-1' is not defined
+sec-1.inc:2: error: syntax error, unexpected SECTION
+sec-2.inc:2: error: 'SEC-2' is not defined
+sec-2.inc:2: error: syntax error, unexpected SECTION
+])
+AT_CHECK([$COMPILE -fformat=cobol85 prog.cob], [0], [],
+[sec-1.inc: in section 'MAIN':
+sec-1.inc:2: warning: optional period used
+sec-2.inc: in section 'SEC-1':
+sec-2.inc:2: warning: optional period used
+])
+AT_CHECK([$COBCRUN_DIRECT ./prog], [0], [], [])
+
+AT_DATA([prog_err.cob], [
+       IDENTIFICATION   DIVISION.
+       PROGRAM-ID.      prog.
+
+       DATA DIVISION.
+       WORKING-STORAGE SECTION.
+       COPY "ok.inc".
+
+       PROCEDURE DIVISION.
+       MAIN SECTION.
+           PERFORM SEC-1.
+           PERFORM SEC-2.
+      * Directive purposefully in Area A
+        REPLACE ==RUNNN== BY ==RUN==.
+           PERFORM SEC-3.
+           STOP RUN
+           COPY "sec-1.inc".
+       COPY "sec-2.inc".
+       SEC-3 SECTION.
+           DISPLAY KO NO ADVANCING.
+])
+
+AT_CHECK([$COMPILE_ONLY -fformat=cobol85 prog_err.cob], [1], [],
+[sec-1.inc: in section 'MAIN':
+sec-1.inc:2: warning: optional period used
+sec-2.inc: in section 'SEC-1':
+sec-2.inc:2: warning: optional period used
+prog_err.cob: in section 'SEC-3':
+prog_err.cob:20: error: 'KO' is not defined
+])
+
+AT_CLEANUP
+
+
+AT_SETUP([statement in Area A])
+AT_KEYWORDS([misc cobol85 area-a])
+
+AT_DATA([prog.cob], [
+       IDENTIFICATION   DIVISION.
+       PROGRAM-ID.      prog.
+       DATA             DIVISION.
+       WORKING-STORAGE  SECTION.
+       01 FLAG          PIC 9 VALUE ZERO.
+       PROCEDURE        DIVISION.
+       MAIN
+       .
+         PERFORM SEC-1
+         IF NOT FLAG = 1
+              DISPLAY "SEC-1 NOT EXECUTED"
+         END-IF
+         STOP RUN
+         .
+       SEC-1 SECTION.
+         MOVE 1 TO FLAG
+         .
+])
+
+AT_CHECK([$COMPILE_ONLY prog.cob], [0], [])
+AT_CHECK([$COMPILE_ONLY -std=cobol85 prog.cob], [1], [],
+[prog.cob:9: error: punctuation in Area A used
+prog.cob: in paragraph 'MAIN':
+prog.cob:10: error: statement in Area A used
+prog.cob:11: error: statement in Area A used
+prog.cob:13: error: syntax error, unexpected END-IF
+prog.cob:14: error: statement in Area A used
+prog.cob:15: error: punctuation in Area A used
+prog.cob: in section 'SEC-1':
+prog.cob:17: error: statement in Area A used
+prog.cob:18: error: punctuation in Area A used
+])
+
+# That error is hard to bypass: -fformat=fixed is needed (and loss of
+# support for missing periods).
+AT_CHECK([$COMPILE_ONLY -std=cobol85 -fnon-word-in-area-a=ok prog.cob], [1], [],
+[prog.cob: in paragraph 'MAIN':
+prog.cob:13: error: syntax error, unexpected END-IF
```

Comment:
@GitMensch In my mind this right here is currently the main issue with this PR, and I think it is hard to overcome with the current approach. Essentially, inserting AREA_A tokens to be able to implement the checks makes us unable to deal with such tokens appearing in the middle of statements, in which case we end up with syntax errors.
But maybe you're ok with this limitation…